### PR TITLE
Add limited Variant binding

### PR DESCRIPTION
**Pull request by gkoz**
I'm somewhat open to suggestions of alternative names for the `VariantType`/`VariantTy` pair.

There's some scarily unsafe stuff there. I might need to add more comments explaining why I think it's sound.

### CODE REVIEW

**Comment by EPashkin** [comment]
Thanks.

File: src/variant.rs [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+
+//! `Variant` binding and helper traits.
+//!
+//! [`Variant`](struct.Variant.html) is an immutable dynamically-typed generic
+//! container. Its type and value are defined at construction and never change.
+//!
+//! `Variant` types are described by [`VariantType`](../struct.VariantType.html)
+//! "type strings".
+//!
+//! Although `GVariant` supports arbitrarily complex types, currently this
```
Comment (GuillaumeGomez) [inline src/variant.rs]: > Although `GVariant` supports arbitrarily complex types, currently this binding is limited to the basic ones:

Instead I propose: "Although `GVariant` supports arbitrarily complex types, this binding is currently limited to the basic ones:"

File: src/variant_type.rs [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+use std::slice;
+
+/// Describes `Variant` types.
+///
+/// The `Variant` type system (based on the D-Bus one) describes types with
+/// "type strings". `VariantType` is an owned immutable type string (you can
+/// think of it as a `Box<str>` statically guaranteed to be a valid type
+/// string), `&VariantTy` is a borrowed one (like `&str`).
+///
+/// Unlike D-Bus signatures, `VariantType` describes the type of exactly one
```
Comment (GuillaumeGomez) [inline src/variant_type.rs]: This sentence sounds a bit weird for me.

**Comment by GuillaumeGomez** [comment]
> I'm somewhat open to suggestions of alternative names for the VariantType/VariantTy pair.

Seems fine for me.

Otherwise excellent work. I wait for the nit and let's merge this!

File: src/variant_type.rs (hunk 2 above)
Comment (gkoz) [inline src/variant_type.rs]: I'll remove it then, we can put complete docs in lgpl-docs later.

**Comment by gkoz** [comment]
I renamed `Variant::to_str` to `get_str` as it's sort of a specialization of `get`. It's plausible that some other `get` variant may appear in the future.

File: src/variant.rs (hunk 1 above)
Comment (gkoz) [inline src/variant.rs]: Done.

**Comment by GuillaumeGomez** [comment]
Agreed. Anything else you want to change before I merge?

**Comment by gkoz** [comment]
Removed a little code duplication. Added `cargo test` to appveyor. If it passes, yep, I'm good.

**Pull request merged**